src/cfg/sc_transformer.cpp

```diff
@@ -39,4 +41,25 @@ std::vector<std::shared_ptr<Node>> parse_to_cfg(antlr4::ANTLRInputStream stream)
     ScTransformer transformer;
     parser.prog()->accept(&transformer);
     return transformer.nodes;
-}
\ No newline at end of file
+}
+
+void remove_function_nodes(ScTransformer& transformer)
+{
+    int nodesToDelete = 0;
+    int functionsToDelete = transformer.functionNodes.size() - 1;
+    int entriesToDelete = 0;
+    for (auto node : transformer.nodes)
+    {
+        FunctionEntryNode* entry = dynamic_cast<FunctionEntryNode*>(node.get());
+        if(entry){
+            if (entry->function_id == "main"){
+                    break;
+                }
+            ++entriesToDelete;
+        }
+        ++nodesToDelete;
+    }
+    transformer.nodes.erase(transformer.nodes.begin(), transformer.nodes.begin() + nodesToDelete);
+    transformer.functionNodes.erase(transformer.functionNodes.begin(),transformer.functionNodes.begin() + functionsToDelete);
+    transformer.entryNodes.erase(transformer.entryNodes.begin(),transformer.entryNodes.begin() + entriesToDelete);
```

Comment:
I think it can be simplified slightly as `functionsToDelete `and `entriesToDelete `should be equal?

Reply:
when i looked at the sizes they were different. Seemed like `functionNodes` contains the functions before any function have been duplicated while `entryNodes` contains all entry nodes after functions have been duplicated.